content/blog/next-level-iac-briding-the-declarative-gap/index.md

```diff
@@ -0,0 +1,164 @@
+---
+title: "Next-level IaC: Bridging the Declarative Gap"
+
+date: 2024-06-07
+draft: false
+
+meta_desc: "Pulumi bridges the worlds of declarative and standard programming using asynchronous semantics."
+meta_image: meta.png
+authors:
+    - troy-howard
+tags:
+    - next-level-iac
+    - typescript
+    - python
+    - architecture
+---
+
+Pulumi stands out in the world of infrastructure-as-code (IaC) for its flexibility and ease of use. This is due to its unique approach of using general purpose programming languages to describe system configuration, rather than flat configuration files or domain-specific languages (DSL). It accomplishes this through a small but brilliant implementation detail that bridges the worlds of declarative and standard programming using asynchronous semantics.
+
+<!--more-->
+
+In Pulumi, you describe your infrastructure in code – real code, not a [DSL][dsl-wiki] or [YAML][yaml-wiki] – using your preferred general purpose programming language You don't have to become a specialist in a niche proprietary declarative language, like Terraform's HCL. Rather, you can reuse your existing programming skills, writing in standard imperative, object-oriented, and even functional language styles, while still gaining all of the benefits of the declarative style that other tools emphasize.
```

Comment:
the first paragraph and this 2nd paragraph repeat a bunch of similar topics. You make a dig at DSL twice. I think we can consolidate.

Reply:
Yep. made some edits to reduce this effect.